weave_utils/models.py

```diff
@@ -59,24 +64,29 @@ def __init__(self, **data):
 
     
     @weave.op()
-    def predict(self, prompt: str):
+    async def predict(self, prompt: str):
         delay = 2
 
         for i in range(self.max_retries):
             try:
-                response = completion(
+                messages = []
+                if self.system_prompt is not None:
+                    messages.append({
```

Comment:
Because we are using litellm, I think they should be supporting the same signature for every model. In case we find an error in future we can add a quick patch to handle.